scripts/lib/ipmi/ipmi_sim_lancontrol

```diff
@@ -32,7 +32,9 @@ link_mac() {
 }
 
 link_subnet() {
-    ifconfig $1 | sed -n -e 's/.*Mask:\([.0-9]*\).*/\1/p'
+    addr=$(ip -o -4 addr list $1 | awk '{print $4}')
+    cidr=${addr##*/}
+    ruby -e "require 'ipaddr'; puts IPAddr.new('255.255.255.255').mask(${cidr})"
```

Comment:
Nice to do it without the need for additional modules. But it is also possible in plain bash (just in case you are considering it as an option to even drop the dependency to ruby):
```
int2ip()
{
    local ui32=$1; shift
    local ip n
    for n in 1 2 3 4; do
        ip=$((ui32 & 0xff))${ip:+.}$ip
        ui32=$((ui32 >> 8))
    done
    echo $ip
}

netmask()
{
    local mask=$((0xffffffff << (32 - $1))); shift
    int2ip $mask
}
```
Taken from: https://stackoverflow.com/questions/15429420/given-the-ip-and-netmask-how-can-i-calculate-the-network-address-using-bash

Reply:
I considered doing that and decided I trusted the builtin ruby module more (and it's only one line of code)